dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileParser.java

```diff
@@ -0,0 +1,177 @@
+ *   <li>Method and Field elements have “className” attribute.
+ * </ul>
+ *
+ * <p>Each type of the element works as a corresponding matcher, such as LinkageErrorMatcher for a
+ * LinkageError element and SourceMatcher for Source element. Given a linkage error, they work as
+ * below:
+ *
+ * <ul>
+ *   <li>A LinkageErrorMatcher matches when all of its child elements match the linkage error.
+ *   <li>A SourceMatcher matches a linkage error when the source class of the error matches one of
+ *       its child elements.
+ *   <li>A TargetMatcher matches a linkage error when the target symbol (class, method, or field) of
+ *       the error matches one of its child elements.
+ *   <li>A PackageMatcher matches the classes that have Java package specified by its name field.
+ *       Prefix to specify child packages.
+ *   <li>A ClassMatcher matches the class specified by its name attribute. ArtifactMatcher,
+ *       PackageMatcher, and ClassMatcher also match methods and fields on their matching classes.
+ *   <li>A MethodMatcher matches method symbol specified by className and name attribute.
+ *   <li>A FieldMatcher matches field symbol specified by className and name attribute.
+ * </ul>
+ */
+class ExclusionFileParser {
+
+  static ImmutableList<LinkageErrorMatcher> parse(Path exclusionFile)
+      throws SAXException, IOException {
+
+    XMLReader xmlReader = XMLReaderFactory.createXMLReader();
+    LinkageErrorMatcherHandler handler = new LinkageErrorMatcherHandler();
+    xmlReader.setContentHandler(handler);
+
+    InputSource inputSource = new InputSource(new FileInputStream(exclusionFile.toFile()));
+    xmlReader.parse(inputSource);
+    return handler.getMatchers();
+  }
+
+  private static class LinkageErrorMatcherHandler extends DefaultHandler {
+
+    private ArrayDeque<SymbolProblemMatcher> stack = new ArrayDeque<>();
+    private ImmutableList.Builder<LinkageErrorMatcher> matchers;
+
+    ImmutableList<LinkageErrorMatcher> getMatchers() {
+      return matchers.build();
+    }
+
+    @Override
+    public void startDocument() {
+      matchers = ImmutableList.builder();
+    }
+
+    private void addMatcherToTop(Object child) throws SAXException {
+      SymbolProblemMatcher parent = stack.peek();
+      if (parent instanceof SourceMatcher && child instanceof SymbolProblemSourceMatcher) {
+        ((SourceMatcher) parent).addMatcher((SymbolProblemSourceMatcher) child);
+      } else if (parent instanceof TargetMatcher && child instanceof SymbolProblemTargetMatcher) {
+        ((TargetMatcher) parent).addMatcher((SymbolProblemTargetMatcher) child);
+      } else {
+        throw new SAXException(
+            "Unexpected parent-child relationship. Parent:" + parent + ", child:" + child);
+      }
+    }
+
+    @Override
+    public void startElement(
+        String namespaceURI, String localName, String qName, Attributes attributes)
+        throws SAXException {
+      switch (localName) {
```

Comment:
you can check that the namespace is empty before the switch statement and error or ignore if it isn't

Reply:
Added that check to throw an error if namespaceURI is not empty.